From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wayne Chang <waynec@nvidia.com>
Date: Fri, 8 Nov 2024 09:25:41 +0000
Subject: [PATCH 159/241] NVIDIA: SAUCE: phy: tegra: xusb: Remove ignore_fuse
 flag

The ignore_fuse flag was previously added as a workaround due to issues
with reading the fuse registers on Tegra264. At the time, the fuse read
functionality was not working correctly. However, with recent improvements,
the fuse reading mechanism is now functioning properly.

As a result, we no longer need to rely on the ignore_fuse flag to bypass
fuse calibration. This commit removes the flag and its associated checks
from the code.

Bug 4933929

Change-Id: I9dfcbefd4c97115902f419a3c00a875834e4bc01
Signed-off-by: Wayne Chang <waynec@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3245449
Reviewed-by: Henry Lin <henryl@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: WK Tsai <wtsai@nvidia.com>
(cherry picked from commit 8cf6aab509a9d803fe2721df67827b2b3d0b9d31)

diff --git a/drivers/phy/tegra/xusb-tegra186.c b/drivers/phy/tegra/xusb-tegra186.c
index 0fdf5dd83f9c..2f2ad4d7a45d 100644
--- a/drivers/phy/tegra/xusb-tegra186.c
+++ b/drivers/phy/tegra/xusb-tegra186.c
@@ -956,30 +956,30 @@ static int tegra186_utmi_phy_power_on(struct phy *phy)
 	value &= ~USB2_OTG_PD_ZI;
 	value |= TERM_SEL;
 
-	if (padctl->soc->ignore_fuse) {
+	if (padctl->is_xhci_iov) {
 		padctl_writel(padctl, value, XUSB_PADCTL_USB2_OTG_PADX_CTL0(index));
-		goto ignore_fuse;
-	}
+		goto skip_fuse_calibration;
+        }
 
 	value &= ~HS_CURR_LEVEL(~0);
-	if (!padctl->is_xhci_iov) {
-		if (usb2->hs_curr_level_offset) {
-			int hs_current_level;
 
-			hs_current_level = (int)priv->calib.hs_curr_level[index] +
-							usb2->hs_curr_level_offset;
+        if (usb2->hs_curr_level_offset) {
+		int hs_current_level;
 
-			if (hs_current_level < 0)
-				hs_current_level = 0;
-			if (hs_current_level > 0x3f)
-				hs_current_level = 0x3f;
+		hs_current_level = (int)priv->calib.hs_curr_level[index] +
+						usb2->hs_curr_level_offset;
 
-			value |= HS_CURR_LEVEL(hs_current_level);
-		} else {
-			value |= HS_CURR_LEVEL(priv->calib.hs_curr_level[index]);
-		}
+		if (hs_current_level < 0)
+			hs_current_level = 0;
+		if (hs_current_level > 0x3f)
+			hs_current_level = 0x3f;
+
+		value |= HS_CURR_LEVEL(hs_current_level);
+	} else {
+		value |= HS_CURR_LEVEL(priv->calib.hs_curr_level[index]);
 	}
 
+
 	padctl_writel(padctl, value, XUSB_PADCTL_USB2_OTG_PADX_CTL0(index));
 
 	value = padctl_readl(padctl, XUSB_PADCTL_USB2_OTG_PADX_CTL1(index));
@@ -989,7 +989,7 @@ static int tegra186_utmi_phy_power_on(struct phy *phy)
 	value |= RPD_CTRL(priv->calib.rpd_ctrl);
 	padctl_writel(padctl, value, XUSB_PADCTL_USB2_OTG_PADX_CTL1(index));
 
-ignore_fuse:
+skip_fuse_calibration:
 	tegra186_utmi_pad_power_on(phy);
 
 	return 0;
@@ -1603,7 +1603,7 @@ tegra186_xusb_padctl_probe(struct device *dev,
 	if (IS_ERR(priv->ao_regs))
 		return ERR_CAST(priv->ao_regs);
 
-	if (!is_xhci_iov && !soc->ignore_fuse) {
+	if (!is_xhci_iov) {
 		err = tegra186_xusb_read_fuse_calibration(priv);
 		if (err < 0)
 			return ERR_PTR(err);
@@ -1844,7 +1844,6 @@ const struct tegra_xusb_padctl_soc tegra264_xusb_padctl_soc = {
 	.poll_trk_completed = true,
 	.trk_hw_mode = true,
 	.supports_lp_cfg_en = true,
-	.ignore_fuse = true,
 };
 EXPORT_SYMBOL_GPL(tegra264_xusb_padctl_soc);
 #endif
diff --git a/drivers/phy/tegra/xusb.h b/drivers/phy/tegra/xusb.h
index 2b4a5f8271e1..09cb98c273ce 100644
--- a/drivers/phy/tegra/xusb.h
+++ b/drivers/phy/tegra/xusb.h
@@ -438,7 +438,6 @@ struct tegra_xusb_padctl_soc {
 	bool trk_update_on_idle;
 	bool supports_lp_cfg_en;
 	bool has_per_pad_term;
-	bool ignore_fuse;
 };
 
 struct tegra_xusb_padctl {
-- 
2.43.0

